### PR TITLE
Cancel scheduled reconnect task during disconnecting ApnsClient

### PR DESCRIPTION
**Pull request by ylgrgyq**
Hi, Dear jchambers, 
I think we should cancel the scheduled reconnect task during disconnecting ApnsClient, otherwise we could have a leaking APNs connection which may never be closed. WDYT?

### CODE REVIEW

**Comment by jchambers** [comment]
Good catch! I think this is important, but is likely to become less relevant after we ship 0.10 (which will include connection pooling and largely discard the notion of "reconnecting"). I think we should *also* fix #407 before 0.10 goes out, too, so I'll create an 0.9.1 branch shortly. When that happens, could I trouble you to re-target this pull request there?

**Comment by ylgrgyq** [comment]
I'd love to, please ping me when you need me to do that. Thanks.

**Comment by jchambers** [comment]
The `v0.9.1` branch is now A Thing; would you mind retargeting/rebasing at your convenience? Thanks!

**Comment by ylgrgyq** [comment]
Done. PTAL @jchambers

**Comment by jchambers** [comment]
(FYI: you can just change the target branch by editing the pull request)

File: pushy/src/main/java/com/relayrides/pushy/apns/ApnsClient.java [hunk 1]
```diff
@@ -469,7 +471,7 @@ public void operationComplete(final ChannelFuture future) throws Exception {
                                 if (ApnsClient.this.reconnectionPromise != null) {
                                     log.debug("Disconnected. Next automatic reconnection attempt in {} seconds.", ApnsClient.this.reconnectDelaySeconds);
 
-                                    future.channel().eventLoop().schedule(new Runnable() {
+                                    scheduledReconnectFuture = future.channel().eventLoop().schedule(new Runnable() {
```
Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsClient.java]: Minor to the point that I almost feel bad asking: could you please qualify this with `ApnsClient.this`?

Comment (ylgrgyq) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsClient.java]: Ah, yes, no problem

Comment (jchambers) [inline pushy/src/main/java/com/relayrides/pushy/apns/ApnsClient.java]: Thanks! Will merge as soon as tests pass.

**Pull request merged**